cluster_management/src/main/java/com/pinterest/rocksplicator/task/BackupTask.java

```diff
@@ -0,0 +1,160 @@
+/// Copyright 2017 Pinterest Inc.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+/// http://www.apache.org/licenses/LICENSE-2.0
+
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+
+//
+// @author kangnanli (kangnanli@pinterest.com)
+//
+
+package com.pinterest.rocksplicator.task;
+
+import com.pinterest.rocksplicator.Utils;
+
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.recipes.locks.InterProcessMutex;
+import org.apache.curator.framework.recipes.locks.Locker;
+import org.apache.helix.task.Task;
+import org.apache.helix.task.TaskResult;
+import org.apache.helix.task.UserContentStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BackupTask extends UserContentStore implements Task {
+  private static final Logger LOG = LoggerFactory.getLogger(BackupTask.class);
+
+  private final String taskCluster;
+  private final String partitionName;
+  private final int backupLimitMbs;
+  private long resourceVersion;
+  private final String job;
+  private final String host;
+  private final int adminPort;
+  private CuratorFramework zkClient;
+  private InterProcessMutex jobVersionMutext;
+
+  public BackupTask(String taskCluster, String resourceName, String partitionName,
+                    int backupLimitMbs, long jobVersion, String job,
+                    String host, int adminPort,
+                    CuratorFramework zkClient) {
+    this.taskCluster = taskCluster;
+    this.partitionName = partitionName;
+    this.backupLimitMbs = backupLimitMbs;
+    this.resourceVersion = jobVersion;
+    this.job = job;
+    this.host = host;
+    this.adminPort = adminPort;
+    this.zkClient = zkClient;
+    this.jobVersionMutext = new InterProcessMutex(zkClient,
+        TaskUtils.getSegmentLockPath(taskCluster, taskCluster, resourceName, jobVersion));
+  }
+
+  /**
+   * Execute the task.
+   * @return A {@link TaskResult} object indicating the status of the task and any additional
+   *         context information that can be interpreted by the specific {@link Task}
+   *         implementation.
+   */
+  @Override
+  public TaskResult run() {
+
+    String dbName = Utils.getDbName(partitionName);
+    String segmentName = partitionName.substring(0, partitionName.lastIndexOf("_"));
+
+    // backup does not need inter-process lock, since jobVersion equal resourceVersion and
+    // jobVersion is already unified at jobConfig level. put into userStore for state map generation
+    try (Locker locker = new Locker(jobVersionMutext)) {
```

Comment:
The comment says we don't need to lock.

Can you explain if we need it or not?

Reply:
removed it. Since task is only responsible execution now, it does not need be responsible to unifying resource version among tasks.